server/src/test/java/com/example/tyfserver/donation/controller/DonationControllerTest.java

```diff
@@ -0,0 +1,256 @@
+import com.example.tyfserver.member.exception.MemberNotFoundException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.Arrays;
+
+import static org.mockito.Mockito.*;
+import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.get;
+import static org.springframework.restdocs.mockmvc.RestDocumentationRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = DonationController.class)
+class DonationControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @MockBean
+    AuthenticationArgumentResolver authenticationArgumentResolver;
+    @MockBean
+    AuthenticationInterceptor authenticationInterceptor;
+    @MockBean
+    private DonationService donationService;
+
+    @Test
+    @DisplayName("/donations - success")
+    public void createDonation() throws Exception {
+        //given
+        DonationRequest request = new DonationRequest("pagename", 1000L);
+        DonationResponse response = new DonationResponse(1L, "name", "message", 1000L);
+        //when
+        when(donationService.createDonation(Mockito.any(DonationRequest.class)))
+                .thenReturn(response);
+        //then
+        mockMvc.perform(post("/donations")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(request)))
+                .andExpect(status().isCreated())
+                .andExpect(jsonPath("donationId").value(1L))
+                .andExpect(jsonPath("name").value("name"))
+                .andExpect(jsonPath("message").value("message"))
+                .andExpect(jsonPath("amount").value(1000L))
+        ;
+    }
+
+    @Test
+    @DisplayName("/donations - 회원을 찾을 수 없음")
+    public void createDonationMemberNotFoundFailed() throws Exception {
+        //given
+        DonationRequest request = new DonationRequest("pagename", 1000L);
+        //when
+        doThrow(new MemberNotFoundException()).when(donationService).createDonation(Mockito.any(DonationRequest.class));
+        //then
+        mockMvc.perform(post("/donations")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(request)))
+                .andExpect(status().isBadRequest())
+                .andExpect(jsonPath("errorCode").value("member-001"))
```

Comment:
에러코드는 예외에서 public 상수를 두고 호출하기로 했던 것 같아! 😃
```suggestion
                .andExpect(jsonPath("errorCode").value(MemberNotFoundException.ERROR_CODE))
``` 
따로 변경점 없으면, 해당 코멘트 우측에 `commit suggestion` 버튼 누르면 될 듯!

Reply:
아 맞다 아 이거 언제 다 바꾸냐